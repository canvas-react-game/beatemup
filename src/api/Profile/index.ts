export * from "./Profile.api";
export { default } from "./Profile.api";