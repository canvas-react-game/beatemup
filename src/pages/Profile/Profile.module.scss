@use "./src/styles/variables.module.scss" as variables;

.formContainer {
  margin: 0 auto auto;
  padding: variables.$spacing_x2;
  max-width: 360px;
}
