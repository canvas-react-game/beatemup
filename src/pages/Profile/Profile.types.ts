export type FormElement = {
    isEdit: boolean;
    setIsEdit: Function;
    form: any;
    onFinish: Function;
};
