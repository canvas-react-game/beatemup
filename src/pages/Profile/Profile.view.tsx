import React, {FC} from 'react';

import Container from "@/components/Container";
import Header from "@/components/Header";

const Profile: FC = () => {
    return (
        <Container>
            <Header/>
        </Container>
    );
};

export default Profile;