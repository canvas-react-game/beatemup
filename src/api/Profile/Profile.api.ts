import { notification } from "antd";

import APIService from "@/services/API";
import { Method } from "@/services/API/API.service";
import { SignUpData, UserInfo } from "@/api/Auth";

const root = "user";

export interface PasswordData {
    oldPassword: string,
    newPassword: string
}

enum ProfileDataType {
    Common = "Common",
    Password = "Password",
}

type DisplayName = {
    display_name: ""
}

class ProfileApi {
    isSuccessfulRequest(response: Response, dataType?: ProfileDataType) {
        const errorMessage = "Отправленные данные не корректны";
        let message = "Данные успешно изменены";
        if (dataType) {
            message = dataType === ProfileDataType.Common ? "Данные профиля успешно изменены"
                : "Пароль успешно изменен";
        }
        switch (response.status) {
            case 200:
                notification.success({ message });
                return true;
            case 400:
                notification.error({ message: errorMessage });
                return false;
            case 401:
                notification.error({ message: "Ошибка авторизации" });
                return false;
            case 500:
                notification.error({ message: "Произошла неизвестная ошибка" });
                return false;
            default:
                return false;
        }
    }

    public async setProfile(data: Omit<SignUpData, "password">): Promise<UserInfo | null> {
        const requestData: Omit<SignUpData, "password"> & DisplayName = {
            ...data, 
            display_name: ""
        }
        const response = await APIService.request(Method.PUT, `${root}/profile`, requestData);
        if (response) {
            const success = this.isSuccessfulRequest(response, ProfileDataType.Common);
            if (success) {
                const result = await response.json();
                return result ?? null;
            }
        }
        return null;
    }

    public async setPassword(data: PasswordData): Promise<boolean> {
        const response = await APIService.request(Method.PUT, `${root}/password`, data);
        if (response) {
            return this.isSuccessfulRequest(response, ProfileDataType.Password);
        }
        return false;
    }

    // todo
    public async setAvatar(data: any): Promise<UserInfo | null> {
        const response = await APIService.request(Method.PUT, `${root}/avatar`, data);
        if (response) {
            const success = this.isSuccessfulRequest(response);
            if (success) {
                const result = await response.json();
                return result ?? null;
            }
        }
        return null;
    }
}

export default new ProfileApi();
