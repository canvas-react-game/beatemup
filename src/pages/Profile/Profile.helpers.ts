import { useCallback, useState } from "react";
import { useForm } from "antd/lib/form/Form";

import { routes } from "@/config/routes/routes";

import Input from "@/components/Input";
import Password from "@/components/Password";

import { ProfileValue } from "./Profile.types";

export const useProfileForm = () => {
  const currentPath = `/#${routes.profile.path}`;

  const onFinish = useCallback(
    () => (values: ProfileValue[]) => console.log(values),
    []
  );
  const onFinishFailed = useCallback(
    () => (errorInfo: any) => console.log("Failed:", errorInfo),
    []
  );

  const [isPreviewVisible, setIsPreviewVisible] = useState<boolean>(false);
  const [isConfirmVisible, setIsConfirmVisible] = useState<boolean>(false);
  const [isEdit, setIsEdit] = useState<boolean>(false);

  const [avatar] = useState<string | undefined>("");
  const [initialValues] = useState({
    name: "Андрей",
    surname: "Иванов",
    email: "test@mail.ru",
    phone: "+77758662255",
    login: "test",
    password: "12345",
  });
  const [form] = useForm();
  const [fields] = useState<ProfileValue[]>([
    {
      name: "name",
      disabled: true,
      required: false,
      placeholder: "Имя",
      component: Input,
    },
    {
      name: "surname",
      disabled: true,
      required: false,
      placeholder: "Фамилия",
      component: Input,
    },
    {
      name: "email",
      disabled: true,
      required: true,
      message: "Введите эл. почту",
      placeholder: "Эл. почта",
      component: Input,
    },
    {
      name: "phone",
      disabled: true,
      required: false,
      placeholder: "Телефон",
      component: Input,
    },
    {
      name: "login",
      disabled: true,
      required: true,
      message: "Введите логин",
      placeholder: "Логин",
      component: Input,
    },
    {
      name: "password",
      disabled: true,
      required: true,
      message: "Введите пароль",
      placeholder: "Пароль",
      component: Password,
    },
  ]);

  return {
    currentPath,
    onFinish,
    onFinishFailed,
    isPreviewVisible,
    setIsPreviewVisible,
    isConfirmVisible,
    setIsConfirmVisible,
    isEdit,
    setIsEdit,
    avatar,
    initialValues,
    form,
    fields,
  };
};
