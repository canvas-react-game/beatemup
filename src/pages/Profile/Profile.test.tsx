import React, { FC } from "react";
import { render } from "@testing-library/react";
import { Router } from "react-router-dom";
import { createMemoryHistory } from "history";

import { Provider } from "react-redux";
import Profile from "@/pages/Profile";
import { makeStore } from "@/helpers/tests/makeStore";
import { compose } from "@/helpers/tests/compose";

describe("Profile page", () => {
    function withRouter(Component: FC): FC {
        const history = createMemoryHistory({ initialEntries: ["/"] });
        return () => (
            <Router history={history}>
                <Component />
            </Router>
        );
    }

    function withStore(Component: FC): FC {
        const store = makeStore();
        return () => (
            <Provider store={store}>
                <Component />
            </Provider>
        );
    }

    test("renders correctly", () => {
        const ProfileComposed = compose(withRouter, withStore)(Profile);
        const tree = render(<ProfileComposed />);

        expect(tree).toMatchSnapshot();
    });

    // Решение бага с тестом ниже
    Object.defineProperty(window, "matchMedia", {
        writable: true,
        value: jest.fn().mockImplementation((query) => ({
            matches: false,
            media: query,
            onchange: null,
            addListener: jest.fn(), // Deprecated
            removeListener: jest.fn(), // Deprecated
            addEventListener: jest.fn(),
            removeEventListener: jest.fn(),
            dispatchEvent: jest.fn(),
        })),
    });
    test("creating component changes profile store state to loading", async () => {
        const store = makeStore();
        const history = createMemoryHistory({ initialEntries: ["/"] });
        const ProfileComposed = () => (<Router history={history}>
            <Provider store={store}>
                <Profile />
            </Provider>
        </Router>);

        render(<ProfileComposed />);

        expect(store.getState().profile["stage"] === "loading").toBe(true);
    });
});
