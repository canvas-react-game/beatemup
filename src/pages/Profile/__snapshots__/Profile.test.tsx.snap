// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Profile page renders correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body
    class="light"
  >
    <div>
      <div
        class="ant-spin ant-spin-lg ant-spin-spinning spinner"
      >
        <span
          aria-label="loading"
          class="anticon anticon-loading anticon-spin ant-spin-dot"
          role="img"
          style="font-size: 72px;"
        >
          <svg
            aria-hidden="true"
            data-icon="loading"
            fill="currentColor"
            focusable="false"
            height="1em"
            viewBox="0 0 1024 1024"
            width="1em"
          >
            <path
              d="M988 548c-19.9 0-36-16.1-36-36 0-59.4-11.6-117-34.6-171.3a440.45 440.45 0 00-94.3-139.9 437.71 437.71 0 00-139.9-94.3C629 83.6 571.4 72 512 72c-19.9 0-36-16.1-36-36s16.1-36 36-36c69.1 0 136.2 13.5 199.3 40.3C772.3 66 827 103 874 150c47 47 83.9 101.8 109.7 162.7 26.7 63.1 40.2 130.2 40.2 199.3.1 19.9-16 36-35.9 36z"
            />
          </svg>
        </span>
      </div>
      <div
        class="loading"
      >
        <div
          class="container"
        >
          <div
            class="container"
          >
            <div
              class="logo"
            >
              <a
                href="/main"
              >
                Logo
              </a>
            </div>
            <div
              class="routesContainer"
            >
              <nav
                class="ulContainer"
              >
                <ul
                  class="ul"
                >
                  <li
                    class="inactive"
                  >
                    <a
                      href="/about"
                    >
                      Об игре
                    </a>
                  </li>
                </ul>
              </nav>
              <div
                class="buttonContainer"
              />
              <button
                class="ant-btn ant-btn-primary"
                type="button"
              >
                <span>
                  Светлая
                </span>
              </button>
            </div>
          </div>
          <div
            class="formContainer"
          >
            <div
              class="ant-space ant-space-vertical"
            >
              <div
                class="ant-space-item"
                style="margin-bottom: 16px;"
              >
                <div
                  class="ant-row ant-row-space-between"
                >
                  <div
                    class="ant-col"
                  >
                    <div
                      class="ant-statistic"
                    >
                      <div
                        class="ant-statistic-title"
                      >
                        Рекорд
                      </div>
                      <div
                        class="ant-statistic-content"
                      >
                        <span
                          class="ant-statistic-content-value"
                        >
                          <span
                            class="ant-statistic-content-value-int"
                          >
                            
                            100
                          </span>
                          
                        </span>
                      </div>
                    </div>
                  </div>
                  <div
                    class="ant-col"
                  >
                    <span
                      class="ant-upload-picture-card-wrapper antUpload"
                    >
                      <div
                        class="ant-upload-list ant-upload-list-picture-card"
                      >
                        <div
                          class="ant-upload ant-upload-select ant-upload-select-picture-card ant-upload-disabled"
                        >
                          <span
                            class="ant-upload ant-upload-disabled"
                            role="button"
                          >
                            <input
                              accept=""
                              style="display: none;"
                              type="file"
                            />
                            <div
                              class="upload"
                            >
                              <span
                                aria-label="plus"
                                class="anticon anticon-plus"
                                role="img"
                              >
                                <svg
                                  aria-hidden="true"
                                  data-icon="plus"
                                  fill="currentColor"
                                  focusable="false"
                                  height="1em"
                                  viewBox="64 64 896 896"
                                  width="1em"
                                >
                                  <defs>
                                    <style />
                                  </defs>
                                  <path
                                    d="M482 152h60q8 0 8 8v704q0 8-8 8h-60q-8 0-8-8V160q0-8 8-8z"
                                  />
                                  <path
                                    d="M176 474h672q8 0 8 8v60q0 8-8 8H176q-8 0-8-8v-60q0-8 8-8z"
                                  />
                                </svg>
                              </span>
                              <div
                                class="button"
                              >
                                Загрузить
                              </div>
                            </div>
                          </span>
                        </div>
                      </div>
                    </span>
                  </div>
                </div>
              </div>
              <div
                class="ant-space-item"
              >
                <form
                  class="ant-form ant-form-vertical"
                  id="profile"
                >
                  <div
                    class="ant-row ant-form-item"
                  >
                    <div
                      class="ant-col ant-form-item-control"
                    >
                      <div
                        class="ant-form-item-control-input"
                      >
                        <div
                          class="ant-form-item-control-input-content"
                        >
                          <input
                            class="ant-input ant-input-disabled"
                            disabled=""
                            id="profile_first_name"
                            placeholder="Имя"
                            type="text"
                            value=""
                          />
                        </div>
                      </div>
                    </div>
                  </div>
                  <div
                    class="ant-row ant-form-item"
                  >
                    <div
                      class="ant-col ant-form-item-control"
                    >
                      <div
                        class="ant-form-item-control-input"
                      >
                        <div
                          class="ant-form-item-control-input-content"
                        >
                          <input
                            class="ant-input ant-input-disabled"
                            disabled=""
                            id="profile_second_name"
                            placeholder="Фамилия"
                            type="text"
                            value=""
                          />
                        </div>
                      </div>
                    </div>
                  </div>
                  <div
                    class="ant-row ant-form-item"
                  >
                    <div
                      class="ant-col ant-form-item-control"
                    >
                      <div
                        class="ant-form-item-control-input"
                      >
                        <div
                          class="ant-form-item-control-input-content"
                        >
                          <input
                            class="ant-input ant-input-disabled"
                            disabled=""
                            id="profile_email"
                            placeholder="Эл. почта"
                            type="text"
                            value=""
                          />
                        </div>
                      </div>
                    </div>
                  </div>
                  <div
                    class="ant-row ant-form-item"
                  >
                    <div
                      class="ant-col ant-form-item-control"
                    >
                      <div
                        class="ant-form-item-control-input"
                      >
                        <div
                          class="ant-form-item-control-input-content"
                        >
                          <input
                            class="ant-input ant-input-disabled"
                            disabled=""
                            id="profile_phone"
                            placeholder="Телефон"
                            type="text"
                            value=""
                          />
                        </div>
                      </div>
                    </div>
                  </div>
                  <div
                    class="ant-row ant-form-item"
                  >
                    <div
                      class="ant-col ant-form-item-control"
                    >
                      <div
                        class="ant-form-item-control-input"
                      >
                        <div
                          class="ant-form-item-control-input-content"
                        >
                          <input
                            class="ant-input ant-input-disabled"
                            disabled=""
                            id="profile_login"
                            placeholder="Логин"
                            type="text"
                            value=""
                          />
                        </div>
                      </div>
                    </div>
                  </div>
                  <div
                    class="ant-row ant-form-item"
                  >
                    <div
                      class="ant-col ant-form-item-control"
                    >
                      <div
                        class="ant-form-item-control-input"
                      >
                        <div
                          class="ant-form-item-control-input-content"
                        >
                          <button
                            class="ant-btn ant-btn-primary ant-btn-block"
                            type="button"
                          >
                            <span>
                              Редактировать
                            </span>
                            <span
                              aria-label="edit"
                              class="anticon anticon-edit"
                              role="img"
                            >
                              <svg
                                aria-hidden="true"
                                data-icon="edit"
                                fill="currentColor"
                                focusable="false"
                                height="1em"
                                viewBox="64 64 896 896"
                                width="1em"
                              >
                                <path
                                  d="M257.7 752c2 0 4-.2 6-.5L431.9 722c2-.4 3.9-1.3 5.3-2.8l423.9-423.9a9.96 9.96 0 000-14.1L694.9 114.9c-1.9-1.9-4.4-2.9-7.1-2.9s-5.2 1-7.1 2.9L256.8 538.8c-1.5 1.5-2.4 3.3-2.8 5.3l-29.5 168.2a33.5 33.5 0 009.4 29.8c6.6 6.4 14.9 9.9 23.8 9.9zm67.4-174.4L687.8 215l73.3 73.3-362.7 362.6-88.9 15.7 15.6-89zM880 836H144c-17.7 0-32 14.3-32 32v36c0 4.4 3.6 8 8 8h784c4.4 0 8-3.6 8-8v-36c0-17.7-14.3-32-32-32z"
                                />
                              </svg>
                            </span>
                          </button>
                        </div>
                      </div>
                    </div>
                  </div>
                </form>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="ant-spin ant-spin-lg ant-spin-spinning spinner"
    >
      <span
        aria-label="loading"
        class="anticon anticon-loading anticon-spin ant-spin-dot"
        role="img"
        style="font-size: 72px;"
      >
        <svg
          aria-hidden="true"
          data-icon="loading"
          fill="currentColor"
          focusable="false"
          height="1em"
          viewBox="0 0 1024 1024"
          width="1em"
        >
          <path
            d="M988 548c-19.9 0-36-16.1-36-36 0-59.4-11.6-117-34.6-171.3a440.45 440.45 0 00-94.3-139.9 437.71 437.71 0 00-139.9-94.3C629 83.6 571.4 72 512 72c-19.9 0-36-16.1-36-36s16.1-36 36-36c69.1 0 136.2 13.5 199.3 40.3C772.3 66 827 103 874 150c47 47 83.9 101.8 109.7 162.7 26.7 63.1 40.2 130.2 40.2 199.3.1 19.9-16 36-35.9 36z"
          />
        </svg>
      </span>
    </div>
    <div
      class="loading"
    >
      <div
        class="container"
      >
        <div
          class="container"
        >
          <div
            class="logo"
          >
            <a
              href="/main"
            >
              Logo
            </a>
          </div>
          <div
            class="routesContainer"
          >
            <nav
              class="ulContainer"
            >
              <ul
                class="ul"
              >
                <li
                  class="inactive"
                >
                  <a
                    href="/about"
                  >
                    Об игре
                  </a>
                </li>
              </ul>
            </nav>
            <div
              class="buttonContainer"
            />
            <button
              class="ant-btn ant-btn-primary"
              type="button"
            >
              <span>
                Светлая
              </span>
            </button>
          </div>
        </div>
        <div
          class="formContainer"
        >
          <div
            class="ant-space ant-space-vertical"
          >
            <div
              class="ant-space-item"
              style="margin-bottom: 16px;"
            >
              <div
                class="ant-row ant-row-space-between"
              >
                <div
                  class="ant-col"
                >
                  <div
                    class="ant-statistic"
                  >
                    <div
                      class="ant-statistic-title"
                    >
                      Рекорд
                    </div>
                    <div
                      class="ant-statistic-content"
                    >
                      <span
                        class="ant-statistic-content-value"
                      >
                        <span
                          class="ant-statistic-content-value-int"
                        >
                          
                          100
                        </span>
                        
                      </span>
                    </div>
                  </div>
                </div>
                <div
                  class="ant-col"
                >
                  <span
                    class="ant-upload-picture-card-wrapper antUpload"
                  >
                    <div
                      class="ant-upload-list ant-upload-list-picture-card"
                    >
                      <div
                        class="ant-upload ant-upload-select ant-upload-select-picture-card ant-upload-disabled"
                      >
                        <span
                          class="ant-upload ant-upload-disabled"
                          role="button"
                        >
                          <input
                            accept=""
                            style="display: none;"
                            type="file"
                          />
                          <div
                            class="upload"
                          >
                            <span
                              aria-label="plus"
                              class="anticon anticon-plus"
                              role="img"
                            >
                              <svg
                                aria-hidden="true"
                                data-icon="plus"
                                fill="currentColor"
                                focusable="false"
                                height="1em"
                                viewBox="64 64 896 896"
                                width="1em"
                              >
                                <defs>
                                  <style />
                                </defs>
                                <path
                                  d="M482 152h60q8 0 8 8v704q0 8-8 8h-60q-8 0-8-8V160q0-8 8-8z"
                                />
                                <path
                                  d="M176 474h672q8 0 8 8v60q0 8-8 8H176q-8 0-8-8v-60q0-8 8-8z"
                                />
                              </svg>
                            </span>
                            <div
                              class="button"
                            >
                              Загрузить
                            </div>
                          </div>
                        </span>
                      </div>
                    </div>
                  </span>
                </div>
              </div>
            </div>
            <div
              class="ant-space-item"
            >
              <form
                class="ant-form ant-form-vertical"
                id="profile"
              >
                <div
                  class="ant-row ant-form-item"
                >
                  <div
                    class="ant-col ant-form-item-control"
                  >
                    <div
                      class="ant-form-item-control-input"
                    >
                      <div
                        class="ant-form-item-control-input-content"
                      >
                        <input
                          class="ant-input ant-input-disabled"
                          disabled=""
                          id="profile_first_name"
                          placeholder="Имя"
                          type="text"
                          value=""
                        />
                      </div>
                    </div>
                  </div>
                </div>
                <div
                  class="ant-row ant-form-item"
                >
                  <div
                    class="ant-col ant-form-item-control"
                  >
                    <div
                      class="ant-form-item-control-input"
                    >
                      <div
                        class="ant-form-item-control-input-content"
                      >
                        <input
                          class="ant-input ant-input-disabled"
                          disabled=""
                          id="profile_second_name"
                          placeholder="Фамилия"
                          type="text"
                          value=""
                        />
                      </div>
                    </div>
                  </div>
                </div>
                <div
                  class="ant-row ant-form-item"
                >
                  <div
                    class="ant-col ant-form-item-control"
                  >
                    <div
                      class="ant-form-item-control-input"
                    >
                      <div
                        class="ant-form-item-control-input-content"
                      >
                        <input
                          class="ant-input ant-input-disabled"
                          disabled=""
                          id="profile_email"
                          placeholder="Эл. почта"
                          type="text"
                          value=""
                        />
                      </div>
                    </div>
                  </div>
                </div>
                <div
                  class="ant-row ant-form-item"
                >
                  <div
                    class="ant-col ant-form-item-control"
                  >
                    <div
                      class="ant-form-item-control-input"
                    >
                      <div
                        class="ant-form-item-control-input-content"
                      >
                        <input
                          class="ant-input ant-input-disabled"
                          disabled=""
                          id="profile_phone"
                          placeholder="Телефон"
                          type="text"
                          value=""
                        />
                      </div>
                    </div>
                  </div>
                </div>
                <div
                  class="ant-row ant-form-item"
                >
                  <div
                    class="ant-col ant-form-item-control"
                  >
                    <div
                      class="ant-form-item-control-input"
                    >
                      <div
                        class="ant-form-item-control-input-content"
                      >
                        <input
                          class="ant-input ant-input-disabled"
                          disabled=""
                          id="profile_login"
                          placeholder="Логин"
                          type="text"
                          value=""
                        />
                      </div>
                    </div>
                  </div>
                </div>
                <div
                  class="ant-row ant-form-item"
                >
                  <div
                    class="ant-col ant-form-item-control"
                  >
                    <div
                      class="ant-form-item-control-input"
                    >
                      <div
                        class="ant-form-item-control-input-content"
                      >
                        <button
                          class="ant-btn ant-btn-primary ant-btn-block"
                          type="button"
                        >
                          <span>
                            Редактировать
                          </span>
                          <span
                            aria-label="edit"
                            class="anticon anticon-edit"
                            role="img"
                          >
                            <svg
                              aria-hidden="true"
                              data-icon="edit"
                              fill="currentColor"
                              focusable="false"
                              height="1em"
                              viewBox="64 64 896 896"
                              width="1em"
                            >
                              <path
                                d="M257.7 752c2 0 4-.2 6-.5L431.9 722c2-.4 3.9-1.3 5.3-2.8l423.9-423.9a9.96 9.96 0 000-14.1L694.9 114.9c-1.9-1.9-4.4-2.9-7.1-2.9s-5.2 1-7.1 2.9L256.8 538.8c-1.5 1.5-2.4 3.3-2.8 5.3l-29.5 168.2a33.5 33.5 0 009.4 29.8c6.6 6.4 14.9 9.9 23.8 9.9zm67.4-174.4L687.8 215l73.3 73.3-362.7 362.6-88.9 15.7 15.6-89zM880 836H144c-17.7 0-32 14.3-32 32v36c0 4.4 3.6 8 8 8h784c4.4 0 8-3.6 8-8v-36c0-17.7-14.3-32-32-32z"
                              />
                            </svg>
                          </span>
                        </button>
                      </div>
                    </div>
                  </div>
                </div>
              </form>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
